packages/admin-ui/src/pages/system/components/Notifications/index.tsx

```diff
@@ -8,33 +8,47 @@ import { InputForm } from "components/InputForm";
 import { ReqStatus } from "types";
 import ErrorView from "components/ErrorView";
 import Ok from "components/Ok";
+import { withToastNoThrow } from "components/toast/Toast";
 
 export default function Notifications() {
   const telegramStatus = useApi.getTelegramStatus();
 
-  const [reqStatus, setReqStatus] = useState<ReqStatus>({});
+  const [reqStatusToken, setReqStatusToken] = useState<ReqStatus>({});
+  const [reqStatusStatus, setReqStatusStatus] = useState<ReqStatus>({});
   const [token, setToken] = useState("");
 
   async function updateTelegramToken() {
     try {
-      setReqStatus({ loading: true });
-      await api.setTelegramToken({ telegramToken: token });
-      setReqStatus({ result: true });
+      setReqStatusToken({ loading: true });
+      await withToastNoThrow(
+        () => api.setTelegramToken({ telegramToken: token }),
+        {
+          message: `Setting telegram token...`,
+          onSuccess: `Updated telegram token`
+        }
+      );
+      setReqStatusToken({ result: true });
       setToken("");
     } catch (e) {
-      setReqStatus({ error: e });
+      setReqStatusToken({ error: e });
       console.error("Error on setTelegramToken", e);
     }
   }
 
   async function updateTelegramStatus(newStatus: boolean) {
     try {
-      setReqStatus({ loading: true });
-      await api.setTelegramStatus({ telegramStatus: newStatus });
+      setReqStatusStatus({ loading: true });
+      await withToastNoThrow(
+        () => api.setTelegramStatus({ telegramStatus: newStatus }),
+        {
+          message: `Switching telegram status...`,
```

Comment:
You can use something like `${newStatus ? "Enabling" : "Disabling"}` so the text is more clear about the action